src/DonorProfiles/resources/js/app/tabs/donation-history/index.js

```diff
@@ -1,9 +1,35 @@
+import axios from 'axios';
+
 // Internal dependencies
 import Content from './content';
+import DashboardContent from './dashboard-content';
+import { store } from './store';
+import { getAPIRoot, getAPINonce } from '../../utils';
+import { setDonations, setQuerying, setCount, setRevenue, setAverage } from './store/actions';
+
+export const registerDonationHistoryTab = () => {
+	const { dispatch } = store;
+
+	dispatch( setQuerying( true ) );
+	axios.get( getAPIRoot() + 'give-api/v2/donor-profile/donations', {
```

Comment:
We should consider extracting the external data logic to a separate location, which would simplify the component and increase test-ability.

Reply:
Because of how tab extensibility is intended, I am opting to avoid centralizing all API logic into one file or place. That said, I'm in agreement that this logic should be extracted to a separate utility object method. I've gone ahead and done so, ref: https://github.com/impress-org/givewp/pull/5455/commits/831447e862f137f92eb69c5ba9ef58803c2f2a1a